core/src/main/scala/zio/jdbc/ZConnectionPool.scala

```diff
@@ -26,6 +26,8 @@ import java.sql.Connection
  */
 final case class ZConnectionPool(transaction: ZLayer[Any, Throwable, ZConnection])
 object ZConnectionPool {
+  private[jdbc] val connectionsCounter = zio.metrics.Metric.counterInt("zio_jdbc_open_connections")
```

Comment:
You need a Gauge here because some metric systems do not allow incrementing by negative numbers.

Otherwise, looks good to merge!